internal/worker/instancepoller/doc.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Package instancepoller provides a worker that periodically polls the
+// cloud provider for instance information and keeps Juju’s domain models
+// up to date.
+//
+// ## Overview
+//
+// The instance poller is responsible for:
```

Comment:
💯 good content.

Please check with @tmihoc on where this content belongs, perhaps `docs/explanation/`?

Reply:
I think this is the best place!